### ISSUE TITLE
[HOLD on #32615] HIGH: Refactor "Room welcome message" to "Room description" (1/2, internal)

### ISSUE THREAD

**Issue by quinthar** [Expensify/App#32953]
*Note: This is a back-end change blocking https://github.com/Expensify/App/issues/32615; will do final cleanup when it's done.*

**Problem:**
We have a featured called "Room welcome messages", which sends a whisper to new users when they join the room.  However, calling it a "welcome message" is proving confusing in practice, especially if we want to show it in more places than just this whisper.

**Solution:**
Refactor the server and client to call this a "Room description", which is more intuitively accurate.  This consists of:

1. add an `updateRoomDescription` API command that updates a `description` rNVP (this is the same rNVP already used for Tasks)
2. return this `description` rNVP for all report types instead of just tasks
3. refactor the client to use this new API and rNVP everywhere the current "room welcome" is.
4. Make sure any workspace member with write access to the room can change this

**Comment by puneetlath** [comment]
Auth PR is in review. https://github.com/Expensify/Auth/pull/9566

Will work on the Web PR next.

**Comment by puneetlath** [comment]
Web PR is here: https://github.com/Expensify/Web-Expensify/pull/40356/files

**Comment by puneetlath** [comment]
This is all done. Just waiting on a web deploy for updating the existing create room command.

**Comment by puneetlath** [comment]
We'll also need to do some cleanup after the front-end is done to remove the old "welcome message" code.

**Comment by quinthar** [comment]
Let's leave this issue open and on HOLD until the front end is done, so we don't forget the cleanup.

**Comment by puneetlath** [comment]
Sounds good thanks.

**Comment by puneetlath** [comment]
Closing this out in favor of the following follow-up issues: 
- https://github.com/Expensify/App/issues/35858
- https://github.com/Expensify/App/issues/35859
- https://github.com/Expensify/App/issues/35860
